crates/sui-types/src/move_package.rs

```diff
@@ -69,6 +76,9 @@ pub struct MovePackage {
     // TODO use session cache
     #[serde_as(as = "BTreeMap<_, Bytes>")]
     module_map: BTreeMap<String, Vec<u8>>,
+
+    /// maps struct/module to a package version where it was first defined
+    type_origin: Option<BTreeMap<ModuleStruct, ObjectID>>,
```

Comment:
This may be an artifact of this being WIP, but I don't think we should make this type `Option`al. Every package should have a type origin table.